File: 14-longest-common-prefix/longest-common-prefix.php
class Solution {

    /**
     * @param String[] $strs
     * @return String
     */
    function longestCommonPrefix($strs) {
        if (empty($strs)) return "";

        $first = $strs[0];

        for ($i = 0; $i < strlen($first); $i++) {
            $char = $first[$i];

            for ($j = 1; $j < count($strs); $j++) {
                // Check bounds + mismatch
                if ($i >= strlen($strs[$j]) || $strs[$j][$i] !== $char) {
                    return substr($first, 0, $i);
                }
            }
        }

        return $first;
    }
}

File: 14-longest-common-prefix/longest-common-prefix.rs
impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
        if strs.is_empty() {
            return "".to_string();
        }

        let first = &strs[0];

        for i in 0..first.len() {
            let ch = first.as_bytes()[i];

            for s in &strs[1..] {
                // bounds check + mismatch
                if i >= s.len() || s.as_bytes()[i] != ch {
                    return first[..i].to_string();
                }
            }
        }

        first.clone()
    }
}